metabase/metabase
# feat(sdk): setup permissions and sandboxing for embedding cli

Closes https://github.com/metabase/metabase/issues/46788

### Description

Asks the user for their multi-tenancy columns, configures the permissions graph with sandboxing enabled, and creates customer collections.

### How to verify

- Run `yarn build-embedding-sdk-cli:watch` in one terminal to build the CLI
- Create an empty React project with Vite
- Run `docker rm -f metabase-enterprise-embedding && yarn add file:../metabase/resources/embedding-sdk && npx @metabase/embedding-sdk-react start`
  - This kills existing Docker container, links the built SDK to your React project, then starts the CLI.
  - This only works if your metabase repo is adjacent to the React project. If not, adjust the `file:..` part
- Verify that group permissions are defined correctly in the Admin Panel
- Follow the flow. Import the `<AnalyticsPage />` React component per the instructions and play with it.

### Work in progress

- [x] Loading the dashboard currently returns `Query requires user attribute customer_id` - I'm working on sampling the tenancy column values, and assigning it to user attributes.
  - This will be assigned via the generated Express server.
- The 3 customer collections doesn't have curated permissions yet - working on this

### Demo

NOTE: this PR does not generate server and component files, those are part of https://github.com/metabase/metabase/pull/47060 which is based on this branch.

[Private Loom](https://www.loom.com/share/64bd4f3173654e298c5c59ee57c82a95?sid=ef6ab729-4353-456e-bb4d-bb326ecb9b1e)

### Checklist
